src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public MateStack() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     public void push(T value) {
+        ensureCapacity();
+        elementData[size] = value;
+        size++;
+    }
 
+    private void ensureCapacity() {
+        if (size == elementData.length) {
+            int newCapacity = elementData.length * 2;
+            Object[] newArray = new Object[newCapacity];
+            System.arraycopy(elementData, 0, newArray, 0, elementData.length);
+            elementData = newArray;
+        }
     }
 
     public T peek() {
-        return null;
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
+
+        return (T) elementData[size - 1];
```

Comment:
Casting with `(T)` is unchecked, which might lead to a `ClassCastException` at runtime if the wrong type is used. Consider using a generic array and the `Array.newInstance` method to create an array of type `T` to store the stack elements.